src/main/java/com/exactpro/th2/uiframework/actions/SendNewOrderSingle.java

```diff
@@ -49,10 +52,20 @@ protected String getName() {
 
 	@Override
 	protected Map<String, String> convertRequestParams(SendNewOrderSingleRequest request) {
-		if (request.getToggleTagsCount() > 0) {
-			return Collections.singletonMap("toggledTags", request.getToggleTagsList().toString());
+		Map<String, String> params = new LinkedHashMap<>();
+		for (SendNewOrderSingleRequest.Tags tags : request.getTagsList()) {
+			var tagNumber = tags.getTagNumber();
+			if (!params.containsKey(tagNumber)) {
+				params.put(tagNumber, tags.getTagValue());
+			} else {
+				int index = 2;
+				String newInd;
+				//noinspection StatementWithEmptyBody
+				while (params.containsKey(newInd = tagNumber + " [" + index++ + "]"));
```

Comment:
Looks like an infinite loop. Add a comment on how it works if it should work like this.